sklearn/linear_model/_stochastic_gradient.py

```diff
@@ -1167,6 +1178,20 @@ class SGDClassifier(BaseSGDClassifier):
     [1]
     """
 
+    _parameter_constraints = {
+        **BaseSGDClassifier._parameter_constraints,
+        "penalty": [StrOptions({"l2", "l1", "elasticnet"}), None],
+        "alpha": [Interval(Real, 0, None, closed="left")],
+        "l1_ratio": [Interval(Real, 0, 1, closed="both")],
+        "power_t": [Interval(Real, None, None, closed="neither")],
```

Comment:
Can we just use `Real` if we don't have any min and max?

Reply:
The difference is that -inf and inf are not in the interval while they are instances of Real, so this is more accurate